### PR TITLE
Improve ETH UX

### LINKED ISSUE #182: Improve sell Ether UX
image

The UX for selling ether is not very good. 
Basically, when you trade we send 2 transactions, without any context.
- 1st tx is the WRAP
- 2nd tx is the trade

It's of to go both in parallel, we can be "optimistic" that the approve will succeed, and then, the order will be taken into account after that. But I think we need to be more explicit about the RESTRICTION we have in our exchange, that we cannot sell ETH.

It would be good to think of ideas about this @biocom , @alfetopito 

Some options (just brainstorming):
- We trade directly in Uniswap, we show a message saying that, disencouraging the user to do it?
- When we select ETH, we use instead WETH, but add somehow the WETH balance and ask the user to wrap first
- When we select ETH, we disable trading. We show a button for opening a WRAP modal, and a link/button to change the selection to WETH instead
- We do as we do now, but instead of sending the two tx, we add a new button that shows [WRAP XXX Eth], being XXX the amount the user want to trade. Then, after the tx is sent, we allow to trade.

### PR DESCRIPTION
**Pull request by W3stside**
Closes #182

Removes ETH trading status but leaves it's WETH capabilities: wrap/unwrap

Provides a modal and explanation section for clarity. Adds 2 options if chainId present:
1. Sell using WETH
2. Wrap ETH

**ETH selected**
Screenshot from 2021-03-04 21-09-53

**Clicking bottom modal**
Screenshot from 2021-03-04 21-09-59

**Clicking trade with WETH**
Screenshot from 2021-03-04 21-10-05

**Clicking Wrap ETH**
Screenshot from 2021-03-04 21-10-20

### MODS
1. `pages/swap`
2. swap/hooks (adds fns)
3. `components/Swap/UnsupportedCurrentcyFooter`

### CODE REVIEW

File: src/custom/pages/Swap/SwapMod.tsx
Comment (alfetopito) [inline src/custom/pages/Swap/SwapMod.tsx:531]: Thanks! This kind of comment makes waaaay easier to review

**Review by alfetopito — APPROVED** [review]
Great PR!

One point, that can/should be addressed separately:

When I chose to BUY `ETH`, the optoins are not perfect (after all, I don't wan't to wrap ETH into WETH if I'm buying ETH), but that's fine:
screenshot_2021-03-04_15-34-27
But when I click to use `Trade with WETH`, it changes the SELL token to `WETH`:
screenshot_2021-03-04_15-34-34
And this is the unwrapping state.

Or maybe you meant to address only sell for now, which is cool.
Approved :+1:

**Comment by W3stside** [comment]
@Mergifyio update

File: src/custom/components/swap/UnsupportedCurrencyFooter/UnsupportedCurrencyFooterMod.tsx
Comment (anxolin) [inline src/custom/components/swap/UnsupportedCurrencyFooter/UnsupportedCurrencyFooterMod.tsx:1]: I really like what you did in this MOD, you basically replaced the hardcoded texts by arbitrary components passed in the props

File: src/custom/state/swap/hooks.ts
Comment (anxolin) [inline src/custom/state/swap/hooks.ts:191]: All this stuff is copy paste,

I believe u didn't create a Mod in this case, becase `src/state/swap/hooks.ts`existed already, and had some copy paste. 

I won't stop the approve, but only if we promise a refactor, where we create the Mod version of this hook, and only override the parts we need to change.

Could you open an issue for this?

**Review by anxolin — COMMENTED** [review]
Awesome! Great work with this one. 
I love the functionality and messages.

I think one problem though, is this message (as you see, I have balance):
Screenshot at Mar 05 12-33-13

Also, maybe for another PR, we should review the styling of this. @biocom do you want to make a proposal? I
Screenshot at Mar 05 12-35-10

File: src/custom/state/swap/hooks.ts
Comment (W3stside) [reply src/custom/state/swap/hooks.ts:191]: done inside #205

**Comment by W3stside** [comment]
@anxolin  @biocom see #208 and #209

**Review by anxolin — APPROVED** [review]

**Pull request merged**
